ui/pages/confirmations/hooks/alerts/useNetworkAndOriginSwitchingAlerts.ts

```diff
@@ -0,0 +1,121 @@
+import { useEffect, useMemo, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { LastInteractedConfirmationInfo } from '../../../../../shared/types/confirm';
+import { Alert } from '../../../../ducks/confirm-alerts/confirm-alerts';
+import { RowAlertKey } from '../../../../components/app/confirm/info/row/constants';
+import { Severity } from '../../../../helpers/constants/design-system';
+import {
+  getLastInteractedConfirmationInfo,
+  setLastInteractedConfirmationInfo,
+} from '../../../../store/actions';
+import { selectNetworkConfigurationByChainId } from '../../../../selectors';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import { useConfirmContext } from '../../context/confirm';
+import { SignatureRequestType } from '../../types/confirm';
+
+const CHAIN_CHANGE_THRESHOLD_MILLISECONDS = 60 * 1000; // 1 Minute
+
+const useNetworkAndOriginSwitchingAlerts = (): Alert[] => {
+  const t = useI18nContext();
+
+  const { currentConfirmation } = useConfirmContext();
+  const { chainId: newChainId = '', id: currentConfirmationId } =
+    currentConfirmation ?? {};
+  const newOrigin =
+    (currentConfirmation as TransactionMeta)?.origin ??
+    (currentConfirmation as SignatureRequestType)?.msgParams?.origin ??
+    '';
+  const network = useSelector((state) => {
+    if (newChainId) {
+      return selectNetworkConfigurationByChainId(state, newChainId);
+    }
+    return undefined;
+  });
+
+  const [lastInteractedConfirmationInfo, updateLastInteractedConfirmationInfo] =
+    useState<LastInteractedConfirmationInfo>();
+
+  useEffect(() => {
+    let isMounted = true;
+    (async () => {
+      const lastConfirmation = await getLastInteractedConfirmationInfo();
+      if (isMounted) {
```

Comment:
Could we early return instead to reduce the nesting?

So we immediately return if `isMounted` is `false`.

And then later return if `isNewId` is `false`?

Reply:
fixed